### ISSUE TITLE
Customized functions

### ISSUE THREAD

**Issue by suntong** [lunet-io/scriban#9]
Hi, 

Apart from the built-in functions, can we define functions in C# and use them in the template, just like what dotliquid can do? 

Thx. 

Excellent project, BTW, and excellent doc & tests too. 
Much better than dotliquid, which I'm converted from. :)

**Comment by xoofx** [comment]
You have multiple import functions on a `ScriptObject`, for example one that takes a [Delegate directly](https://github.com/lunet-io/scriban/blob/8d8dbd1c426870d86ca97dae035f422a777910be/src/Scriban/Runtime/ScriptObjectExtensions.cs#L244)

Then after, instead of passing directly your .NET object, you need to use the `TemplateContext` directly to customize the global context object. like how it is done when using [template.Render(Object)](https://github.com/lunet-io/scriban/blob/8d8dbd1c426870d86ca97dae035f422a777910be/src/Scriban/Template.cs#L103))

Basically something like this:

```csharp
var template = Template.Parse(...)
var model = new { xxx = "tes" };
var scriptObject = new ScriptObject();
scriptObject.Import(model);
scriptObject.Import(model, "myfunction", new Func<string>() => "Hello");

var context = new TemplateContext();
context.PushGlobal(scriptObject);
template.Render(context);
context.PopGlobal();
```

**Comment by suntong** [comment]
Sorry I am still not fully understand that [`Delegate` function](https://github.com/lunet-io/scriban/blob/8d8dbd1c426870d86ca97dae035f422a777910be/src/Scriban/Runtime/ScriptObjectExtensions.cs#L244), as my understanding of C# is that `delegate` in lower case, as [in here `public delegate`](https://github.com/lunet-io/scriban/blob/8d8dbd1c426870d86ca97dae035f422a777910be/src/Scriban/Runtime/ScriptObjectExtensions.cs#L23). Where is that `Delegate` keyword defined? I did [some search within scriban](https://github.com/search?l=&p=1&q=Delegate+repo%3Alunet-io%2Fscriban&ref=advsearch&type=Code&utf8=%E2%9C%93) but didn't find it. Thx.

**Comment by xoofx** [comment]
Sorry, the code above was probably not compiling, but something like this should work:

```
// Create on scriptObject.myfunction a function that return "Hello"
scriptObject.Import("myfunction", new Func<string>( ()=> "Hello"));
```
You should take the time to dig into C#, as I can't really afford to provide support for this. 😉

**Comment by suntong** [comment]
yeah, sure, as long as I can get it working. Thanks a lot!

**Comment by suntong** [comment]
Now working like a charm. Thanks. 

BTW, just FTR, the link above to `template.Render(Object))` was wrong, should be [this one](https://github.com/lunet-io/scriban/blob/8d8dbd1c426870d86ca97dae035f422a777910be/src/Scriban/Template.cs#L67).

**Comment by xoofx** [comment]
the link was not wrong. I just linked to the method you were originally using to show what it was calling behind 😉 (content that I copy pasted after)
Glad that this is all working 👍

**Comment by suntong** [comment]
Ah, gotya, I was referring to the code `template.Render(context)` from above.
